main/services/devon-instances/devon-instances.service.ts

```diff
@@ -1,99 +1,150 @@
-              .then((results) => {
-                for (const result of results) {
-                  instanceCount = instanceCount + result;
-                }
-                resolve(instanceCount);
-              })
-              .catch((error) => {
-                console.log(error);
-                reject(error);
-              });
-          }
-        }
-      );
-    });
-    return dirReader;
-  }
-
-  getInstances(instancepath: string): Promise<number> {
-    const devonInstances = new Promise<number>((resolve, reject) => {
-      fs.readdir(path.resolve(instancepath, 'workspaces'), (error, files) => {
-        if (error) reject(resolve(0));
-        if (files) {
-          resolve(files.length);
-        }
-      });
-    });
-    return devonInstances;
-  }
-
-  getAllUserCreatedDevonInstances(): Promise<DevonfwConfig> {
-    let paths = [];
-    const instances: DevonfwConfig = { distributions: [] };
-    const instancesDirReader = new Promise<DevonfwConfig>((resolve, reject) => {
-      fs.readFile(
-        path.resolve(process.env.USERPROFILE, '.devon', 'ide-paths'),
-        'utf8',
-        (err, data) => {
-          if (err) reject('No instances find out');
-          if (data) {
-            paths = data.split('\n');
-            for (let singlepath of paths) {
-              if (singlepath) {
-                if (platform() === 'win32') {
-                  singlepath = singlepath.replace('/', '');
-                  singlepath = singlepath.replace('/', ':/');
-                  singlepath = singlepath.replace(/\//g, path.sep);
-                }  
-                const instance: IdeDistribution = {
-                  id: singlepath,
-                  ideConfig: {
-                    version: '',
-                    basepath: singlepath,
-                    commands: path.resolve(singlepath, 'scripts', 'command'),
-                    workspaces: path.resolve(singlepath, 'workspaces'),
-                  },
-                };
-                exec(
-                  'devon -v',
-                  { cwd: path.resolve(singlepath, 'scripts') },
-                  (_: unknown, stdout: string) => {
-                    instance.ideConfig.version = stdout;
-                  }
-                );
-                instances.distributions.push(instance);
-              }
-            }
-            resolve(instances);
-          }
-        }
-      );
-    });
-    return instancesDirReader;
-  }
-}
+import fs from 'fs';
+import path from 'path';
+import platform from 'os';
+import {
+  DevonfwConfig,
+  IdeDistribution,
+} from '../../models/devonfw-dists.model';
+import * as child from 'child_process';
+import { ProjectDetails } from '../../models/project-details.model';
+
+const exec = child.exec;
+
+export class DevonInstancesService {
+  private devonFilePath = path.resolve(
+    platform.homedir(),
+    '.devon',
+    'projectinfo.json'
+  );
+
+  getAvailableDevonIdeInstances(): Promise<number> {

```

Comment:
Could you split this code in mutiple functions?

Reply:
Hi @jambulud ,
I pushed the changes. Please review it and let me know in case any issue

Thanks